ols/app/models/config.py

```diff
@@ -214,35 +214,42 @@ def validate_yaml(self) -> None:
                 )
 
 
-class OpenAIConfig(BaseModel):
-    """Configuration specific to OpenAI provider."""
+class ProviderSpecificConfig(BaseModel):
+    """Base class with common provider specific configurations."""
 
     url: AnyHttpUrl  # required attribute
+    token: Optional[Any] = None
+    api_key: Optional[str] = None
 
 
-class AzureOpenAIConfig(BaseModel):
+class OpenAIConfig(ProviderSpecificConfig):
+    """Configuration specific to OpenAI provider."""
+
+    credentials_path: str  # required attribute
+
+
+class AzureOpenAIConfig(ProviderSpecificConfig):
     """Configuration specific to Azure OpenAI provider."""
 
-    url: AnyHttpUrl  # required attribute
     deployment_name: str  # required attribute
+    credentials_path: Optional[str] = None
```

Comment:
So because it is optional here, we can't have `credentials_path` in parent, right?

Reply:
exactly